lib/logflare/gen_singleton/watcher.ex

```diff
@@ -49,6 +49,11 @@ defmodule Logflare.GenSingleton.Watcher do
   end
 
   @impl true
+  def handle_info({:DOWN, ref, :process, _pid, reason}, state)
+      when ref == state.monitor_ref and reason in [:normal, :shutdown] do
+    {:stop, :normal, state}
+  end
+
```

Comment:
Any issues with this change? 

I couldn't see any reason we'd need Watcher to restart a process that has exited normally (and don't want UserMetricPoller to be restarted if it wants to shutdown).

Reply:
Had to make another change so opened #2778